publish-wrapper

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import socket
+import subprocess
+import sys
+
+from task import github
+from task import sink
+
+sys.dont_write_bytecode = True
+
+HOSTNAME = socket.gethostname().split(".")[0]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Log and publish into github')
+    parser.add_argument('--publish', help='Publish results centrally to a sink',
+                        default=os.environ.get("TEST_PUBLISH"))
+    parser.add_argument('--test-name', help='Test name',
+                        default=os.environ.get("TEST_NAME", "tests"))
+    parser.add_argument('--test-project', help="Project which is being tested",
+                        default=os.environ.get("TEST_PROJECT", ""))
+    parser.add_argument('--base-project', help="Project in which the tested PR is opened",
+                        default=os.environ.get("BASE_PROJECT"))
+    parser.add_argument('--test-branch', help="Branch to be tested - if differs to PR base branch",
+                        default=os.environ.get("BASE_BRANCH", ""))
+    parser.add_argument('--revision', help="Revision of the PR head",
+                        default=os.environ.get("TEST_REVISION"))
+    parser.add_argument('--context', help="The context or type of integration tests to run",
+                        default=os.environ.get("TEST_CONTEXT"))
+    parser.add_argument('command', help="Command to wrap")
+    opts = parser.parse_args()
+
+    if not opts.test_project:
+        opts.test_project = opts.base_project
+
+    print(opts)
+    for opt in ["publish", "base_project", "revision", "context"]:
+        if getattr(opts, opt) is None:
+            parser.error("the following argument is required: {0}".format(opt))
+
+    try:
+        task = PullTask()
+        ret = task.run(opts)
+    except RuntimeError as ex:
+        ret = str(ex)
+
+    if ret:
+        sys.stderr.write("log-wrapper: {0}\n".format(ret))
```

Comment:
This whole ret= thing could be simplified. Drop the `return 0`, and move this error handling right into the `except:` above.

Reply:
the `ret` is also set up in `try` as well as in `except`. I think it would make it just more complicated?
```
     try:                                                                        
          task = PullTask()                                                       
          ret = task.run(opts)                                                    
          if ret:                                                                 
              sys.stderr.write("log-wrapper: {0}\n".format(ret))                  
          return 0                                                                
      except RuntimeError as ex:                                                  
          ret = str(ex)                                                           
          if ret:                                                                 
              sys.stderr.write("log-wrapper: {0}\n".format(ret))                  
          return 1 
```